Add data dictionary plugin and orchestration to docker build

mhocc opened the pull request:
Added `./build_scripts/build_orchestrator.sh` based on `build_explorer.sh` with the ARGS changed to those for the orchestrator

Created new ARGS based on the `MDM_EXPLORER` with the prefix `NHSD_DD_ORCHESTRATION`

Added a new `NHSD_ORCHESTRATION_PORT` to use in place of `MDM_PORT` which will need to be set in the environment when the image is built.

Added a script to bring the container up using the `develop` branch rather than default `main`

Resolves #1

mhocc committed:
Add data dictionary plugin and orchestration to the build

mhocc committed:
Add data dictionary plugin and orchestration to the build

- Added scripts to build from the develop branch
- Made the Postgres server accessible from the host

Commit:
Added scripts to start the containers on the develop branch

mhocc committed:
gh-1: Renamed the up script indicating it is for the develop branch

pjmonks commented on develop-up.sh:
```diff
@@ -0,0 +1,24 @@
+
+export MDM_PORT=80
+export MDM_APPLICATION_COMMIT=develop
+export MDM_UI_COMMIT=develop
```
Setting the correct environment variables seems important. As I was testing on my machine, I was learning about the use of the `.env` file which I think is more useful. If we set these values in the `.env` file, then the `docker compose` commands will automatically pick up those environment variables. 

The `.env` file is the default, but others can be used (I've spotted an irrelevant `.env.cd` file, which I think represents a continuous delivery setup for standard Mauro). The way to use different `.env` files is to pass them in with a command argument:

```bash
docker compose --env-file .env.other ...
```

I think this might help with defining the other environments - Test, Training and Live. I'll make further issues for this.

Action for you @mhocc - could you update the `.env` file please and put these port and commit values into it?

mhocc replied:
Done

pjmonks commented on docker-compose.yml:
```diff
@@ -9,17 +8,23 @@ services:
                 MDM_UI_COMMIT: "${MDM_UI_COMMIT}"
-                ADDITIONAL_PLUGINS: ""
-                MDM_UI_THEME_NAME: "default"
+                ADDITIONAL_PLUGINS: "uk.ac.ox.softeng.maurodatamapper.plugins:mdm-plugin-nhs-data-dictionary:2.0.0-SNAPSHOT"
```
Good start, and I've tested the NHS plugin endpoints in my container which all seem to work. We might need to add additional plugins later, but I'm not sure what those will be yet - this is fine for now.

pjmonks commented on docker-compose.yml:
```diff
@@ -9,17 +8,23 @@ services:
+                ADDITIONAL_PLUGINS: "uk.ac.ox.softeng.maurodatamapper.plugins:mdm-plugin-nhs-data-dictionary:2.0.0-SNAPSHOT"
+                MDM_UI_THEME_NAME: "nhs-digital"
                 CACHE_BURST: "${CACHE_BURST}"
                 MDM_EXPLORER_API_ENDPOINT: "http://localhost:${MDM_PORT}/api"
```
We could probably remove references to the MDM Explorer, I don't think NHSE will be using it.

mhocc replied:
The explorer UI has been replaced with the orchestration UI

pjmonks commented on mauro-data-mapper/Dockerfile:
```diff
@@ -8,14 +8,19 @@ LABEL org.opencontainers.image.authors="Oliver Freeman <oliver.freeman@bdi.ox.ac
 ARG DEPLOY_MDM_EXPLORER=0
 ARG MDM_EXPLORER_VERSION="1.0.0-SNAPSHOT"
 ARG MDM_EXPLORER_BUILD_HOME=/opt/mdm-explorer/dist
 ARG MDM_EXPLORER_API_ENDPOINT="http://localhost:${MDM_PORT}/api"
```
Other references to MDM Explorer which could probably be removed (including anything else in the repo you spot which is related).

mhocc replied:
All explorer references and sources removed

pjmonks reviewed (changes requested):
This is working great @mhocc 👍 I've also tested ingesting the data dictionary and that all works too.

It took me a while to understand what was happening, but it's actually quite simple. I've captured notes as I was installing everything so I'll update the README with relevant instructions, I've created issue #2 for that purpose.

There are just a few things I think could be cleaned up before I merge it.

mhocc committed:
gh-1: Remove the MDM explorer UI

Commit:
gh-1: Update build environment, remove explorer UI

pjmonks committed:
gh-1 Adjustments to Docker build

- Add a tag name for the local development image
- Restore copy of mdm-ui and nhsd-datadictionary-orchestration to correct webapps folders. Keeping them in separate folders will map them to URLs correctly.

pjmonks reviewed (approved):
I fixed the copy of the web apps back to use sub-folders, a change you made ended up only being able to access the Orchestrator UI. Other than that it all works great.
